src/Discovery/index.tsx

```diff
@@ -73,21 +72,55 @@ const DiscoveryWithMDSBackend: React.FC<{
     throw new Error('Could not find configuration for Discovery page. Check the portal config.');
   }
 
+  // Downloads and processes studies in two seperate batches
+  // to improve load time & usability
+  // Initially uses a smaller batch to load interface quickly
+  // Then a batch with all the studies
+  const [numberOfBatchesLoaded, setNumberOfBatchesLoaded] = useState(0);
+  const expectedNumberOfTotalBatches = 2;
+  // if loading with both unregistered and registered studies, load 5 for each (10 total)
+  const numberOfStudiesForSmallerBatch = isEnabled('studyRegistration') ? 5 : 10;
+  const numberOfSmallBatchesToBeLoaded = isEnabled('studyRegistration') ? 2 : 1;
+  const totalNumberOfStudiesFromSmallBatches = numberOfStudiesForSmallerBatch * numberOfSmallBatchesToBeLoaded;
+  const numberOfStudiesForAllStudiesBatch = 2000;
+
   useEffect(() => {
+    // If batch loading is Enabled, update the numberOfBatchesLoaded to enable calling of different batch sizes
```

Comment:
is there a way to disable batch loading?

Reply:
Good catch. That was initially scoped, but then we decided to have this functionality for all instances of the discovery page. The comment has been updated to: 

`    // Update the numberOfBatchesLoaded to enable calling of different batch sizes with different parameters`